frontend/src/components/MapView/LeftPanel/layersPanel/MenuItem/MenuSwitch/SwitchItem/LayerDownloadOptions.tsx

```diff
@@ -138,6 +139,78 @@ function LayerDownloadOptions({
     handleDownloadMenuClose();
   };
 
+  // Helper function to escape special XML characters
+  const escapeXml = (str: string): string => {
+    return str.replace(/</g, '&lt;').replace(/>/g, '&gt;');
+  };
+
+  // Helper function to generate QML content from legend
+  const generateQmlContent = (
+    legend: LegendDefinitionItem[],
+    opacity: number = 1,
+  ): string => {
+    let qml = `<!DOCTYPE qgis PUBLIC 'http://mrcc.com/qgis.dtd' 'SYSTEM'>
+<qgis hasScaleBasedVisibilityFlag="0" styleCategories="AllStyleCategories">
+    <pipe>
+        <rasterrenderer opacity="${opacity}" alphaBand="-1" band="1" classificationMin="-1" classificationMax="inf" type="singlebandpseudocolor">
+            <rasterTransparency />
+            <rastershader>
+                <colorrampshader colorRampType="DISCRETE" classificationMode="1" clip="0">`;
+    // Add color entries for each legend item
+    legend.forEach((item, index) => {
+      const label = item.label
+        ? escapeXml(item.label as string)
+        : item.value.toString();
+
+      // TEMPORARY: shift the value index by 1 to account for the 0 value
+      // and match the QML style format. See https://github.com/WFP-VAM/prism-app/pull/1161
+      const shouldShiftIndex =
+        legend[0].value === 0 &&
+        ((legend[0].label as string)?.includes('<') ||
+          (legend[1].label as string)?.includes('<') ||
+          (legend[1].label as string)?.includes('-') ||
+          (legend[1].label as string)?.includes(' to '));
```

Comment:
@wadhwamatic I added this logic. I looked through the layers.json to find the different patterns that could lead us to shift the legends. Not 100% I got everything right but we can adapt it in the future if we notice that a specific layer is off